service/src/main/java/com/derpgroup/derpwizard/resource/AlexaResource.java

```diff
@@ -0,0 +1,126 @@
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.amazon.speech.json.SpeechletRequestEnvelope;
+import com.amazon.speech.json.SpeechletResponseEnvelope;
+import com.amazon.speech.speechlet.SpeechletRequest;
+import com.amazon.speech.speechlet.SpeechletResponse;
+import com.amazon.speech.ui.SimpleCard;
+import com.amazon.speech.ui.SsmlOutputSpeech;
+import com.derpgroup.derpwizard.configuration.MainConfig;
+import com.derpgroup.derpwizard.voice.alexa.AlexaRequestType;
+import com.derpgroup.derpwizard.voice.alexa.AlexaSkillsKitUtil;
+
+/**
+ * REST APIs for requests generating from Amazon Alexa
+ *
+ * @author Eric
+ * @since 0.0.1
+ */
+@Path("/alexa")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class AlexaResource {
+
+  public AlexaResource(MainConfig config, Environment env) {}
+
+  /**
+   * Generates a welcome message.
+   *
+   * @return The message, never null
+   */
+  @POST
+  public SpeechletResponseEnvelope doAlexaRequest(SpeechletRequestEnvelope request){
+    if(request == null || request.getRequest() == null){
+      throw new RuntimeException("Missing request body."); //TODO: create AlexaException
+    }
+    SpeechletRequest sr = request.getRequest();
+    AlexaRequestType requestType = AlexaSkillsKitUtil.getRequestType(sr);
+    
+    switch(requestType){
+    case LaunchRequest:
+      return doLaunchRequest(request);
+    case IntentRequest:
+      return doIntentRequest(request);
+    case SessionEndedRequest:
+      return doSessionEndedRequest(request);
+      default: 
+        throw new RuntimeException("Unknown request type."); //TODO: create AlexaException
+    }
+  }
+  
+  protected SpeechletResponseEnvelope doLaunchRequest(SpeechletRequestEnvelope request){
+    SpeechletResponseEnvelope response = new SpeechletResponseEnvelope();
+    response.setVersion("0.0.1");
+    
+    SsmlOutputSpeech outputSpeech = new SsmlOutputSpeech();
+    outputSpeech.setSsml("<speak><p><s>Hi. This is Alexa, speaking on behalf of DerpWizard.</s></p></speak>");
+    
+    SimpleCard card = new SimpleCard();
+    card.setContent("Hi. This is Alexa, speaking on behalf of DerpWizard.");
+    card.setTitle("Alexa + DERPWizard");
+
+    SpeechletResponse sr = new SpeechletResponse();
+    sr.setOutputSpeech(outputSpeech);
+    sr.setCard(card);
+    response.setResponse(sr);
+    response.setSessionAttributes(request.getSession().getAttributes());
+    
+    return response;
+  }
+  
+  protected SpeechletResponseEnvelope doIntentRequest(SpeechletRequestEnvelope request){
+    SpeechletResponseEnvelope response = new SpeechletResponseEnvelope();
+    response.setVersion("0.0.1");
+    
+    SsmlOutputSpeech outputSpeech = new SsmlOutputSpeech();
+    outputSpeech.setSsml("<speak><p><s>Hi. This is Alexa, speaking on behalf of DerpWizard.</s></p></speak>");
+    
+    SimpleCard card = new SimpleCard();
```

Comment:
Long-term we should define some sort of `VisualOutput` interface. For now it would only be used to push Card objects with the Alexa response, but in the future we could use them for other mobile apps.